### ISSUE TITLE
Reconfigure ballot entry so speaker selection is first

### ISSUE THREAD

**Issue by philipbelesky** [czlee/tabbycat#188]
I think the ballot entry pages would be more efficient if the speaker orders were the first bit of data entry (in an area of the form dedicated just to this purpose) and subsequent ballots only allowed number entry. Mostly because it allows for faster keyboard entry not having to switch between TAB/arrow keys/numbers, and means it is faster to 'scan' the ballot only having to worry about one information set at a time.

**Comment by philipbelesky** [comment]
Possibly as a part of this: there could be some added javascript that eliminates speaker options as they are selected, so that in teams of three with three options, you would not need to set the final speaker (it would be done by process of elimination). Potentially some tradeoff in data validation for efficiency though?

**Comment by czlee** [comment]
Should be easy enough, I moved all the tab index setting to a dedicated function because it was getting complicated:
https://github.com/czlee/tabbycat/blob/master/debate/forms.py#L298-L334

Re the second part, I think I implemented that for NZ Easters so it should already work for teams of two. I think I decided against doing so for teams of three partly in a nod to that data validation trade-off.

**Comment by czlee** [comment]
I think I wrote the current tab order because it was easier for me, but I'm not averse to doing all speaker names first.

**Comment by philipbelesky** [comment]
I've done a quick draft of this in the commit above; did you want to check it out CZ? It's a bit awkward in terms of naming and class reuse but seems to work well enough.

**Comment by czlee** [comment]
Wait, sorry, I'm confused. I thought all we were doing was talking about tab order, in which case it should be a simple fix to `BallotSetForm._set_tab_indices()`. If we're looking at rearranging the form layout I'm much less of a fan—I think the form is more intuitive when it looks like a scoresheet, and it's not very hard for a pair of human eyes to filter names from numbers.

**Comment by czlee** [comment]
To rearrange tab orders, change the lines at https://github.com/czlee/tabbycat/blob/master/debate/forms.py#L310-L315:

``` python
        for side, pos in self.SIDES_AND_POSITIONS:
            order.append(self._fieldname_speaker(side, pos))
            order.append(self._fieldname_score(self.adjudicators[0], side, pos))

        for adj, side, pos in itertools.product(self.adjudicators[1:], self.SIDES, self.POSITIONS):
            order.append(self._fieldname_score(adj, side, pos))
```

to:
``` python
        for side, pos in self.SIDES_AND_POSITIONS:
            order.append(self._fieldname_speaker(side, pos))

        for adj, side, pos in itertools.product(self.adjudicators, self.SIDES, self.POSITIONS):
            order.append(self._fieldname_score(adj, side, pos))
```

For people using the form for the first time, I don't really think speed is important, I'd much rather that the form look intuitive. For people doing speed entry as tab assistants, I'd tend to just encourage them to use the keyboard only, in which case the form layout shouldn't matter so long as it's visually parsable, only tab order.

**Comment by philipbelesky** [comment]
Ah, you're right the visuals don't really matter as long as the tabbing works properly. Reverted/fixed in 66ddc34e59d1b030f214ab78b9d4083591d594e8
